base/c.jl

```diff
@@ -212,6 +212,11 @@ containsnul(s::String) = containsnul(unsafe_convert(Ptr{Cchar}, s), sizeof(s))
 containsnul(s::AbstractString) = '\0' in s
 
 function unsafe_convert(::Type{Cstring}, s::Union{String,AbstractVector{UInt8}})
+    if !(s isa String)
+        # make sure s is null-terminated
+        push!(s, 0x00)
+        pop!(s)
```

Comment:
Just curious – how does this ensure that `s` is null-terminated? A `0x00` is pushed but popped immediately, so the content of `s` should be the same as at the beginning, no?

Reply:
A pop does not reverse the memory effects of a push, just the change to the length